n2survey/lime/survey.py

```diff
@@ -280,6 +306,115 @@ def get_responses(
 
         return responses
 
+    def __getitem__(
+        self, key: Union[pd.Series, pd.DataFrame, str, tuple]
+    ) -> Union["LimeSurvey", pd.DataFrame]:
+        """Retrieve or slice the responses DataFrame
+
+        Args:
+            key (pd.Series, pd.DataFrame, str, or tuple): A key for filter or get_responses method
+
+        Returns:
+            LimeSurvey or pd.DataFrame: A copy of LimeSurvey instance with filtered responses
+                DataFrameor or sliced responses DataFrame
+        """
+        # A bool-valued Series e.g. survey.responses["A3"] == "A5"
+        if isinstance(key, (pd.Series, pd.DataFrame)):
+            return self.filter(key)
+        # A question id e.g. "A3"
+        elif isinstance(key, str):
+            return self.get_responses(key)
+        # Two args e.g. survey.responses["A3"] == "A5", "B1"
+        # or 1:10, "B1"
+        elif isinstance(key, tuple):
+            rows, columns = key
+            return self.filter(rows).get_responses(columns)
+        else:
+            raise SyntaxError(
+                "Argument must be of type pd.Series, pd.DataFrame, str, or tuple."
```

Comment:
This message might confuse a user. Remember, the user does not call it as a function rather it is called as a slicing with `[]`. So, the reference to `argument` is not clear. It would be more helpful for a user to have some list of examples of what is allowed